packages/app-lixi-api/src/modules/account/follow.resolver.ts

```
@@ -0,0 +1,335 @@
+import _ from 'lodash';
+import { findManyCursorConnection } from '@devoxa/prisma-relay-cursor-connection';
+import { PageAccountEntity } from 'src/decorators/pageAccount.decorator';
+
+const pubSub = new PubSub();
+
+@Resolver(() => FollowAccount)
+@UseFilters(GqlHttpExceptionFilter)
+export class FollowResolver {
+  constructor(private logger: Logger, private prisma: PrismaService, @I18n() private i18n: I18nService) {}
+
+  @Subscription(() => FollowAccount)
+  followAccountCreated() {
+    return pubSub.asyncIterator('followAccountCreated');
+  }
+
+  // Follow Account
+  @Query(() => FollowAccount)
+  @UseGuards(GqlJwtAuthGuard)
+  async checkIsFollowedAccount(
+    @PageAccountEntity() account: Account,
+    @Args('address', { type: () => String }) address: string
+  ) {
+    const accountQuery = await this.prisma.account.findFirst({
+      where: { address: address }
+    });
+
+    if (!account || !accountQuery) {
+      const couldNotFindAccount = await this.i18n.t('page.messages.couldNotFindAccount');
+      throw new VError.WError(couldNotFindAccount);
+    }
+
+    const existData = await this.prisma.followAccount.findFirst({
+      where: {
+        followerAccountId: accountQuery.id,
+        followingAccountId: account.id
+      }
+    });
+
+    const result = {
+      ...existData,
+      isFollowed: existData ? true : false
+    };
+
+    return result;
+  }
+
+  @Query(() => FollowAccountConnection)
+  async allFollowers(
+    @Args() { after, before, first, last }: PaginationArgs,
+    @Args({ name: 'query', type: () => Number, nullable: true })
+    query: number,
+    @Args({
+      name: 'orderBy',
+      type: () => FollowAccountOrder,
+      nullable: true
+    })
+    orderBy: FollowAccountOrder
+  ) {
+    const result = await findManyCursorConnection(
+      paginationArgs =>
+        this.prisma.followAccount.findMany({
+          where: {
+            OR: !query
```

Why do we need to handle the case of an invalid "query"? We can just throw an error or return an empty array if the followingId or accountId needs to be checked is invalid.